File: frontend/lib/models/quiz_question_model.dart
class QuizQuestionModel {
  int? id;
  int? quizId;

  String question;

  String optionA;
  String optionB;
  String optionC;
  String optionD;

  String correctAnswer;

  QuizQuestionModel({
    this.id,
    this.quizId,
    required this.question,
    required this.optionA,
    required this.optionB,
    required this.optionC,
    required this.optionD,
    required this.correctAnswer,
  });

  factory QuizQuestionModel.fromJson(
      Map<String, dynamic> json) {
    return QuizQuestionModel(
      id: json["id"],
      quizId: json["quizId"],
      question: json["question"],
      optionA: json["optionA"],
      optionB: json["optionB"],
      optionC: json["optionC"],
      optionD: json["optionD"],
      correctAnswer: json["correctAnswer"],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "quizId": quizId,
      "question": question,
      "optionA": optionA,
      "optionB": optionB,
      "optionC": optionC,
      "optionD": optionD,
      "correctAnswer": correctAnswer,
    };
  }
}

File: backend/src/models/quizQuestion.model.js
const { DataTypes } = require("sequelize");
const sequelize = require("../config/db");

const QuizQuestion = sequelize.define(
  "QuizQuestion",
  {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
    },

    quizId: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },

    question: {
      type: DataTypes.TEXT,
      allowNull: false,
    },

    optionA: {
      type: DataTypes.STRING(255),
      allowNull: false,
    },

    optionB: {
      type: DataTypes.STRING(255),
      allowNull: false,
    },

    optionC: {
      type: DataTypes.STRING(255),
      allowNull: false,
    },

    optionD: {
      type: DataTypes.STRING(255),
      allowNull: false,
    },

    correctAnswer: {
      type: DataTypes.ENUM("A", "B", "C", "D"),
      allowNull: false,
    },
  },
  {
    tableName: "QuizQuestions",
    timestamps: true,
  }
);

module.exports = QuizQuestion;